### PR TITLE
test: update ostree_ref of fedora during upgrade image building

### PR DESCRIPTION
**Pull request by mcattamoredhat**
This PR is a suggested workaround for https://github.com/virt-s1/rhel-edge/issues/5958

### CODE REVIEW

**Comment by henrywang** [comment]
@mcattamoredhat Could you please check the remote ostree repo in vm? There might be a `fedora-iot` in the vm but it doesn't point to our ostree repo. If yes after your checking, we should do like https://github.com/virt-s1/rhel-edge/blob/b458cf0fc9fc07825effb8f023ef5f9d9d3a5e12/ostree-raw-image.sh#L948

**Comment by mcattamoredhat** [comment]
It seems that VM does not have "${PROD_REPO_URL}" as ostree remote repository configured. 
```
[simple@localhost ~]$ ostree remote list --show-urls 
fedora-iot    https://ostree.fedoraproject.org/iot
```

That repo does not have the branches we need `OSTREE_REF="test/fedora/x/${ARCH}/iot"`
```
[simple@localhost ~]$ ostree remote refs fedora-iot
fedora-iot:fedora/29/aarch64/iot
fedora-iot:fedora/29/armhfp/iot
fedora-iot:fedora/29/x86_64/iot
fedora-iot:fedora/30/aarch64/iot
fedora-iot:fedora/30/armhfp/iot
fedora-iot:fedora/30/x86_64/iot
fedora-iot:fedora/devel/aarch64/iot
fedora-iot:fedora/devel/armhfp/iot
fedora-iot:fedora/devel/x86_64/iot
fedora-iot:fedora/rawhide/aarch64/iot
fedora-iot:fedora/rawhide/armhfp/iot
fedora-iot:fedora/rawhide/x86_64/iot
fedora-iot:fedora/stable/aarch64/iot
fedora-iot:fedora/stable/armhfp/iot
fedora-iot:fedora/stable/x86_64/iot
```

I've added "${PROD_REPO_URL}" manually for testing purposes 
```
[simple@localhost ~]$ ostree remote refs fedora-iot-2
fedora-iot-2:fedora/39/x86_64/iot
fedora-iot-2:test/fedora/x/x86_64/iot
[simple@localhost ~]$ ostree remote list --show-urls 
fedora-iot    https://ostree.fedoraproject.org/iot
fedora-iot-2  http://192.168.100.1/repo
[simple@localhost ~]$ cat /etc/ostree/remotes.d/fedora-iot-2.conf 
[remote "fedora-iot-2"]
url=http://192.168.100.1/repo
gpg-verify=false
```
In that case, rebase seems to work:
```
[simple@localhost ~]$ sudo -S rpm-ostree rebase fedora-iot-2:test/fedora/x/x86_64/iot
116 metadata, 104 content objects fetched; 10821 KiB transferred in 1 seconds; 30.2 MB content written
Staging deployment... done
Added:
  libmetalink-0.1.3-32.fc39.x86_64
  wget-1.21.3-7.fc39.x86_64
Changes queued for next boot. Run "systemctl reboot" to start a reboot
```

**Comment by henrywang** [comment]
@mcattamoredhat Thanks for doing this. I think the best way to do that (to work with RHEL or Centos) is delete the default fedora-iot and add new fedora-iot, but the new one points to our own ostree repo.

**Comment by mcattamoredhat** [comment]
@henrywang Thanks! Makes sense. I will continue the tests following your suggestion.

File: ostree-simplified-installer.sh [hunk 1]
```diff
@@ -1087,6 +1092,8 @@ sudo composer-cli blueprints delete rebase > /dev/null
 
 # Rebase to new REF.
 greenprint "🗳 Rebase to new ostree REF"
+sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" ${BLUEPRINT_USER}@${PUB_KEY_GUEST_ADDRESS} "echo ${EDGE_USER_PASSWORD} |sudo -S ostree remote delete ${REF_PREFIX}"
```
Comment (henrywang) [inline ostree-simplified-installer.sh:1095]: This might for Fedora only.

Comment (mcattamoredhat) [reply ostree-simplified-installer.sh]: Done! Let's see what happens.

File: ostree-simplified-installer.sh [hunk 2]
```diff
@@ -1385,6 +1394,11 @@ sudo composer-cli compose delete "${COMPOSE_ID}" > /dev/null
 sudo composer-cli blueprints delete upgrade > /dev/null
 
 greenprint "🗳 Upgrade ostree image/commit"
+# Update default Fedora's repository
+if [[ "$ID" == fedora ]]; then
+    sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" ${BLUEPRINT_USER}@${PUB_KEY_GUEST_ADDRESS} "echo ${EDGE_USER_PASSWORD} |sudo -S ostree remote delete ${REF_PREFIX}"
```
Comment (henrywang) [inline ostree-simplified-installer.sh:1399]: It's not `PUB_KEY_GUEST_ADDRESS` but `ROOT_CERT_GUEST_ADDRESS`.

Comment (mcattamoredhat) [reply ostree-simplified-installer.sh]: Sure, done. Thanks!

File: ostree-simplified-installer.sh [hunk 3]
```diff
@@ -1385,6 +1394,11 @@ sudo composer-cli compose delete "${COMPOSE_ID}" > /dev/null
 sudo composer-cli blueprints delete upgrade > /dev/null
 
 greenprint "🗳 Upgrade ostree image/commit"
+# Update default Fedora's repository
+if [[ "$ID" == fedora ]]; then
+    sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" ${BLUEPRINT_USER}@${ROOT_CERT_GUEST_ADDRESS} "echo ${EDGE_USER_PASSWORD} |sudo -S ostree remote delete ${REF_PREFIX}"
+    sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" ${BLUEPRINT_USER}@${ROOT_CERT_GUEST_ADDRESS} "echo ${EDGE_USER_PASSWORD} |sudo -S ostree remote add --no-gpg-verify ${REF_PREFIX} https://ostree.fedoraproject.org/iot"
```
Comment (henrywang) [inline ostree-simplified-installer.sh:1400]: It's not `https://ostree.fedoraproject.org/iot`, but `${PROD_REPO_URL}`.

**Review by henrywang — CHANGES_REQUESTED** [review]
I have my comment.

File: ostree-simplified-installer.sh (hunk 3 above)
Comment (mcattamoredhat) [reply ostree-simplified-installer.sh]: Thanks! Done.

**Review by henrywang — APPROVED** [review]
LGTM. Thanks @mcattamoredhat!

**Pull request merged**